### PR TITLE
Give Allowance if necessary

### PR DESCRIPTION
**Pull request by fleupold**
This PR adds allowance for the GP allowance manager contract to spend the sell token on behalf of the user if necessary. Note, that the used PK needs to have some ETH for gas in order to set those approvals.

### Testplan
Successful trade: https://protocol-rinkeby.dev.gnosisdev.com/api/v1/orders/0xc30c1b3bf942d045b53d4374ec6da206e903aa7e3705e2235e680e4e52a6e001e0b3700e0aadcb18ed8d4bff648bc99896a18ad1605a28d0

To repro:
- `export PK=<your key with some rinkeby funds from the token list below (e.g. WETH)>`
- `yarn hardhat trade --token-list-url https://raw.githubusercontent.com/compound-finance/token-list/master/compound.tokenlist.json --network rinkeby` (note that this can sporadically fail as not all tokens in that list have enough liquidity).
- check status of returned UID via `/api/v1/orders/<uid>`

### CODE REVIEW

File: src/make_trade.ts [hunk 1]
```diff
@@ -163,3 +176,18 @@ async function signOrder(
...
+
+async function giveAllowanceIfNecessary(
+  sellToken: TokenInfo,
+  sellAmount: BigNumber,
+  trader: SignerWithAddress,
+  allowanceManager: string,
+  ethers: HardhatEthersHelpers
+) {
+  const erc20 = await toERC20(sellToken.address, ethers);
+  const allowance = await erc20.allowance(trader.address, allowanceManager);
```
Comment (alfetopito) [inline src/make_trade.ts:188]: Don't you have to convert the return to BigNumber?

Comment (nlordell) [reply src/make_trade.ts:188]: If I'm not mistaken, `ethers` will do this by default when ABI decoding the return data.

File: src/make_trade.ts [hunk 2]
```diff
@@ -17,6 +20,8 @@ import fetch from "node-fetch";
 import { Api } from "./api";
 import { Chain, ChainUtils, selectRandom, toERC20 } from "./utils";
 
+const MAX_ALLOWANCE = BigNumber.from(2).pow(256).sub(1);
```
Comment (nlordell) [inline src/make_trade.ts:23]: ubernit:
```suggestion
const MAX_ALLOWANCE = ethers.constants.MaxUint256;
```

**Review by nlordell — APPROVED** [review]
Looks good.

**Review by alfetopito — APPROVED** [review]

File: src/make_trade.ts [hunk 3]
```diff
@@ -67,6 +72,15 @@ export async function makeTrade(
     } for ${buyAmount} of ${buyToken.name} with a ${fee.toString()} fee`
   );
 
+
+  await giveAllowanceIfNecessary(
```
Comment (fleupold) [inline src/make_trade.ts:76]:
```suggestion
  await giveAllowanceIfNecessary(
```

**Pull request merged**